pkg/clusteragent/admission/mutate/autoinstrumentation/env_vars.go

```diff
@@ -75,21 +101,32 @@ func (e envVar) mutateContainer(c *corev1.Container) error {
 	index := slices.IndexFunc(c.Env, func(ev corev1.EnvVar) bool {
 		return ev.Name == e.key
 	})
-	if index < 0 {
-		c.Env = append(c.Env, corev1.EnvVar{
-			Name:  e.key,
-			Value: e.valFunc(""),
-		})
+
+	var found bool
+	var priorEnvVar corev1.EnvVar
+	if index >= 0 {
+		priorEnvVar = c.Env[index]
+		found = true
+	}
+
+	nextEnvVar, err := e.nextEnvVar(priorEnvVar, found)
+	if err != nil {
+		return err
+	}
+
+	if found {
+		c.Env[index] = nextEnvVar
 	} else {
-		if c.Env[index].ValueFrom != nil {
-			return fmt.Errorf("%q is defined via ValueFrom", e.key)
-		}
-		c.Env[index].Value = e.valFunc(c.Env[index].Value)
+		c.Env = appendOrPrepend(nextEnvVar, c.Env, e.prepend)
```

Comment:
I'm having trouble following this logic. Might be Friday-brain. I'll take another look, but I'm convinced we can make this simpler.